shared/message.h

```diff
@@ -142,23 +163,21 @@ struct message_fusemagic {
     int primary_magic_id;
     int secondary_magic_id;
     nibashared::magic magic;
-    bool success = false;
 };
 
-struct message_reordermagic {
+struct message_reordermagic : public base_message<message_reordermagic> {
     // use the same message for equipping & unequipping magics
     const cmdtype type = cmdtype::reordermagic;
 
     message_reordermagic() = default;
-    message_reordermagic(std::vector<int>&& equipped_magic_ids);
+    explicit message_reordermagic(std::vector<int> &&equipped_magic_ids);
     bool validate(const nibashared::sessionstate &session);
     nlohmann::json create_response();
     nlohmann::json create_request();
     void merge_response(const nlohmann::json &j);
```

Comment:
Since we now use base_* methods? These validate, create and merge are no longer used?

Reply:
the `base_*` methods actually call these

Reply:
I see. Basically, it's the same static polymorphism as introduced in the wiki page. The template parameter is passed into the base class at compile time. There is a Capture Conversion concept in Java which seems to be very similar. Thanks for bringing this interesting feature here.